### PR TITLE
✨ Implement event api

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

1. Add interfaces and types for the Event API.
2. Rename `getEventsForProcessModel` to `getStartEventsForProcessModel`

PR: #30

## How can others test the changes?

Use the contracts to implement the Event API.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Fixing Security Issues   | `:lock:`             | 🔒     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️      |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️      |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/data_models/event/event.ts [hunk 1]
```diff
@@ -1,10 +1,40 @@
...
+  /**
+   * The corresponding id of the Correlation holding the event.
+   */
+  public correlationId: string;
+  /**
+   * The corresponding id of the ProcessModel.
+   */
+  public processModelId: string;
+  /**
+   * The id of the ProcessInstance.
```
Comment (robpal94) [inline src/data_models/event/event.ts]:
```suggestion
   * Optional: The id of the ProcessInstance.
```

File: src/data_models/event/event_list.ts [hunk 2]
```diff
@@ -4,5 +4,8 @@ import {Event} from '../index';
  * Contains a list of events.
  */
 export interface EventList {
+  /**
+   * The event list.
+   */
```
Comment (robpal94) [inline src/data_models/event/event_list.ts]: I don't think this comment is needed here

File: src/data_models/event/event_trigger_payload.ts [hunk 3]
```diff
@@ -1,4 +1 @@
-/**
- * Describes the payload with which an event can be triggered.
- */
-export type EventTriggerPayload = any; // TODO: Define EventTrigger-body
+export type EventTriggerPayload = any;
```
Comment (robpal94) [inline src/data_models/event/event_trigger_payload.ts]: Was the `//todo` resolved? 😁

File: src/imanagement_api.ts [hunk 4]
```diff
@@ -176,9 +176,76 @@ export interface IManagementApi {
    * @returns              A Promise, which resolves with the retrieved events,
    *                       or rejects an error, in case the request failed.
    *                       This can happen, if the ProcessModel was not found,
-  *                        or the user is not authorized to see the it.
+   *                        or the user is not authorized to see the it.
    */
-  getEventsForProcessModel(identity: IIdentity, processModelId: string): Promise<EventList>;
+  getStartEventsForProcessModel(identity: IIdentity, processModelId: string): Promise<EventList>;
+
+    /**
```
Comment (robpal94) [inline src/imanagement_api.ts]:
```suggestion
   /**
```

File: src/imanagement_api.ts [hunk 5]
```diff
@@ -176,9 +176,76 @@ export interface IManagementApi {
...
+   * Retrieves a list of all triggerable events belonging to an instance of a
+   * specific ProcessModel.
```
Comment (robpal94) [inline src/imanagement_api.ts]: `Retrieves a list of all triggerable events, which belongs to an instance of a specific ProcessModel`.

File: src/data_models/event/event_trigger_payload.ts (hunk 3 above)
Comment (ElRaptorus) [reply src/data_models/event/event_trigger_payload.ts]: Yep. 
This will remain as it is, for now.

File: src/data_models/event/event.ts (hunk 1 above)
Comment (ElRaptorus) [reply src/data_models/event/event.ts]: This comment is actually pretty pointless anyway. I'll just remove it.

File: src/imanagement_api.ts (hunk 5 above)
Comment (ElRaptorus) [reply src/imanagement_api.ts]: Would be incorrect.

**Pull request merged**